### DISCUSSION TITLE
PolymorphicTypeInvalid if primary key is the same...

### DISCUSSION THREAD

**Discussion by jedie** [jazzband/django-polymorphic#806] [Q&A]
It's possible to create objects with same primary keys. But it will be raise into:

`polymorphic.models.PolymorphicTypeInvalid: ContentType XY for <class 'FooBar'> #XY does not point to a subclass!`

So there is not `unique_together` contraints between the primary key and `polymorphic_ctype`, isn't it?

**Comment by jedie** [comment]
I demonstrate this with: https://github.com/jedie/django-polymorphic/commit/adb038b1948be467d0b61fc8341c01c775d2cc08

A test output looks like: https://travis-ci.org/django-polymorphic/django-polymorphic/jobs/623225731#L320-L338

**Comment by bckohan** [comment]
@jedie This may have been a django issue in <=2.x? It looks like you were "downcasting" the original row by creating child rows, using pks that reference the parent. When I do this on 4.2+ I get unique constraint errors - presumably because django is trying to create parent rows using the taken pk.

Even if you were still allowed to do this, an exception would no longer be raised on fetch thanks to #752 and #785

We also now offer a convenience function for "downcasting". See [create_from_super](https://django-polymorphic.readthedocs.io/en/latest/api/polymorphic.managers.html#polymorphic.managers.PolymorphicManager.create_from_super).
